bundles/org.openhab.binding.xmltv/src/main/java/org/openhab/binding/xmltv/internal/handler/ChannelHandler.java

```diff
@@ -0,0 +1,249 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.xmltv.internal.handler;
+
+import static org.openhab.binding.xmltv.internal.XmlTVBindingConstants.*;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.types.RawType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.xmltv.internal.configuration.XmlChannelConfiguration;
+import org.openhab.binding.xmltv.internal.jaxb.Icon;
+import org.openhab.binding.xmltv.internal.jaxb.MediaChannel;
+import org.openhab.binding.xmltv.internal.jaxb.Programme;
+import org.openhab.binding.xmltv.internal.jaxb.Tv;
+import org.openhab.binding.xmltv.internal.jaxb.WithLangType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ChannelHandler} is responsible for handling information
+ * made available in regard of the channel and current program
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@NonNullByDefault
+public class ChannelHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(XmlTVHandler.class);
+
+    @NonNullByDefault({})
```

Comment:
Please place null annotations between the modifiers and type. See also the [Null Annotations Conventions](https://www.openhab.org/docs/developer/development/conventions.html#null-annotations).